### DISCUSSION TITLE
How to bind to events in XAML?

### DISCUSSION THREAD

**Discussion by DevAM-Tools** [AvaloniaUI/Avalonia#13496] [Q&A]
I would like to bind an Action or a Command to the DoubleTapped event. (comparable to the click event of a Button) 

Is there a (general) way to bind to events in XAML?

**Comment by DarthMazut** [comment]
You could use [Avalonia.Xaml.Behaviors](https://github.com/AvaloniaUI/Avalonia.Xaml.Behaviors) package. 
Here is some example:
https://github.com/AvaloniaUI/Avalonia.Xaml.Behaviors/blob/46e516a0a11a349528bc19051fbcaafaf7b29609/samples/BehaviorsTestApplication/Views/Pages/InvokeCommandActionView.axaml#L24-L30

I don't think it's possible to bind event per se, but behaviours are good way to trigger e.g. ViewModel commands from View.

**Comment by timunie** [comment]
you cannot bind events, you need to add them in code behind or use a behavior if you really need it in the ViewModel. 

> **NOTE**
> If you use Rider, you will get the code behind autogenerated.
